src/QMCWaveFunctions/lcao/CuspCorrectionConstruction.cpp

```diff
@@ -0,0 +1,256 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//                    Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "QMCWaveFunctions/lcao/CuspCorrectionConstruction.h"
+
+namespace qmcplusplus
+{
+
+  // Modifies orbital set lcwc
+  void applyCuspCorrection(const Matrix<CuspCorrectionParameters> &info, int num_centers,
+                            int orbital_set_size, ParticleSet& targetPtcl, ParticleSet& sourcePtcl,
+                            LCAOrbitalSetWithCorrection& lcwc, const std::string &id)
+  {
+    typedef QMCTraits::RealType RealType;
+
+    LCAOrbitalSet phi = LCAOrbitalSet(lcwc.myBasisSet);
+    phi.setOrbitalSetSize(lcwc.OrbitalSetSize);
+    phi.BasisSetSize = lcwc.BasisSetSize;
+    phi.setIdentity(false);
+
+    LCAOrbitalSet eta = LCAOrbitalSet(lcwc.myBasisSet);
+    eta.setOrbitalSetSize(lcwc.OrbitalSetSize);
+    eta.BasisSetSize = lcwc.BasisSetSize;
+    eta.setIdentity(false);
+
+
+    std::vector<bool> corrCenter(num_centers, "true");
+
+    //What's this grid's lifespan?  Why on the heap?
+    LogGrid<RealType>* radial_grid = new LogGrid<RealType>;
+    radial_grid->set(0.000001, 100.0, 1001);
```

Comment:
Where do these magic numbers come from?

Reply:
(Wondering if we should start flagging magic numbers consistently in the source. For sure we will be bitten one day.)

Reply:
I created an issue #1400 for discussion or trouble shooting magic number issues. So we don't need to delay the merge of this PR.